### ISSUE TITLE
Pivot Date dimension months option performance

### ISSUE THREAD

**Issue by damyanpetev** [IgniteUI/igniteui-angular#12406]
## Description  
In the Pivot Date Dimension if you enable dimensions per month (`months` option) there's a very noticeable drop in performance. It seems to be to do with the date formatting which can be optimized.

 * igniteui-angular version: 15.0.x
 * browser: Chromium

## Steps to reproduce  

1. Open https://stackblitz.com/edit/angular-f3pmyr?file=src%2Fapp%2Fpivot-grid%2Fpivot-datasource%2Fpivot-data-selector-sample.component.ts
2. Tigger a re-render - e.g. resize the Rows Dimensions column or disable and re-enable the Date Dimension
3. Enable dimensions per month (`months` option) 
4. Repeat step 2 and observe the difference

## Result  
With dimensions per month:
![pivot-with-month](https://user-images.githubusercontent.com/3198469/205087858-0750b0ab-286c-4e8c-a4f5-3e9cf7c8c801.gif)
From quick digging a major chunk of this can be traced to the month member function of the dimention:
![image](https://user-images.githubusercontent.com/3198469/205090767-643a4af8-fa10-435d-b58e-66763ebd2acb.png)
Basically this:
https://github.com/IgniteUI/igniteui-angular/blob/551d27e7f032ef51fae168296fa3beb9ce3bd198/projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid-dimensions.ts#L109-L111
Which relies on calling `toLocaleString` with formatting, which internally is likely creating `Intl` instances every time - explains the high GC share. Than can become costly pretty quick with data, but should also be able to be avoided pretty easily by caching a single `Intl.DateTimeFormat` instance to do the formatting for all

## Expected result  
Speed comparable to without dimensions per month:
![pivot-without-month](https://user-images.githubusercontent.com/3198469/205087871-5bba1d6b-f1e6-4052-87e4-859f1ebd1e8c.gif)
